l1-contracts/src/core/messagebridge/Registry.sol

```diff
@@ -14,28 +14,46 @@ import {DataStructures} from "@aztec/core/libraries/DataStructures.sol";
  * @notice Keeps track of important information for L1<>L2 communication.
  */
 contract Registry is IRegistry {
-  // TODO(rahul) - https://github.com/AztecProtocol/aztec-packages/issues/526
-  // Need to create a snashot of addresses per version!
-
-  DataStructures.L1L2Addresses public addresses;
-
-  function setAddresses(address _rollup, address _inbox, address _outbox) public {
-    addresses = DataStructures.L1L2Addresses(_rollup, _inbox, _outbox);
-  }
-
-  function getL1L2Addresses() external view override returns (DataStructures.L1L2Addresses memory) {
-    return addresses;
+  uint256 public latestVersionNumber;
+  mapping(uint256 version => DataStructures.RegistrySnapshot snapshot) snapshots;
+  DataStructures.RegistrySnapshot latestSnapshot;
+
+  // todo: this function has to be permissioned.
+  function upgrade(address _rollup, address _inbox, address _outbox) public {
+    latestVersionNumber++;
+    DataStructures.RegistrySnapshot memory newSnapshot =
+      DataStructures.RegistrySnapshot(_rollup, _inbox, _outbox, block.number);
+    latestSnapshot = newSnapshot;
+    snapshots[latestVersionNumber] = newSnapshot;
```

Comment:
`latestVersionNumber == 0` will never be updated with this. If this is intended, would be nice to have as a comment somewhere. Also, instead of latest, it could be `currentShapshot` and `currentVersion`, just to signal that it is not just the latest, but also the active one. People are often using a counter instead of current number so looks a little fun with the `latestVersionNumber++` but think it is fine.

Reply:
fixed!